src/Action/Admin/Login.php

```diff
@@ -49,6 +35,8 @@ public function showLoginForm(Request $request, Response $response) : Response
 
         $this->renderer->appendLocalCss('login', __DIR__ . '/../../../public/css/page-login.css');
 
+        $response = $response->withAddedHeader('Cache-Control', 'no-cache');
```

Comment:
Login page never showed error messages because it was always served from cache 🤦

Reply:
haha nice find